src/domain/usecases/CopyDataSetsOUUserCase.ts

```diff
@@ -61,29 +62,26 @@ export class CopyDataSetsOUUserCase {
 function compare(dataSet1: DataSet, dataSet2: DataSet, replace: boolean): boolean {
     const OUs1ToDiff = getDataSetWithSortedOUs(dataSet1);
     const OUs2ToDiff = getDataSetWithSortedOUs(dataSet2);
-    let differences;
 
     if (replace) {
-        differences = _.isEqual(OUs1ToDiff, OUs2ToDiff);
+        return _.isEqual(OUs1ToDiff, OUs2ToDiff);
     } else {
-        const filtered = OUs2ToDiff.filter((ou) => {
+        const filtered = OUs2ToDiff.filter(ou => {
             return ou.id === OUs1ToDiff.find(ou2 => ou2.id === ou.id)?.id;
-        })
-        differences = _.isEqual(OUs1ToDiff, filtered)
+        });
+        return _.isEqual(OUs1ToDiff, filtered);
     }
-
-    return differences;
 }
 
 function getDataSetWithSortedOUs(dataSet: DataSet): DataSet["organisationUnits"] {
     return sortById(dataSet.organisationUnits);
 }
 
 function mergeDataSetOUs(orgDataset: DataSet, destDataSet: DataSet): DataSet {
-    const mergedOUs: DataSet["organisationUnits"] = _.uniqBy(
-        [...orgDataset.organisationUnits, ...destDataSet.organisationUnits],
-        "id"
-    );
+    const mergedOUs: DataSet["organisationUnits"] = _.uniq([
+        ...orgDataset.organisationUnits,
+        ...destDataSet.organisationUnits,
+    ]);
```

Comment:
`_.uniqBy` was ok (objects built separately are never equal in JS), I meant: `_.uniqBy(xs, x => x.id)`

Reply:
Ok, changed it to _.uniqBy. But in this case _.uniq compares the object values (uniq -> SameValueZero -> SameValueNonNumber), so, the id value.